cmd/gardener-controller-manager/app/gardener_controller_manager.go

```diff
@@ -416,11 +417,17 @@ func determineGardenerIdentity() (*gardencorev1alpha1.Gardener, string, error) {
 
 	// If running inside a Kubernetes cluster (as container) we can read the container id from the proc file system.
 	// Otherwise generate a random string for the gardenerID
-	if cgroup, err := ioutil.ReadFile("/proc/self/cgroup"); err == nil {
-		splitByNewline := strings.Split(string(cgroup), "\n")
-		splitBySlash := strings.Split(splitByNewline[0], "/")
-		gardenerID = splitBySlash[len(splitBySlash)-1]
-	} else {
+	if cGroupFile, err := os.Open("/proc/self/cgroup"); err == nil {
```

Comment:
what is the motivation behind reading the file via Open?

Reply:
We can prevent to read the whole file to memory at once which could probably cause a resource exhaustion (result from static code analysis).